scene/resources/texture.h

```diff
@@ -77,6 +83,7 @@ class Texture : public Resource {
 	virtual void draw_rect(RID p_canvas_item, const Rect2 &p_rect, bool p_tile = false, const Color &p_modulate = Color(1, 1, 1), bool p_transpose = false, const Ref<Texture> &p_normal_map = Ref<Texture>()) const;
 	virtual void draw_rect_region(RID p_canvas_item, const Rect2 &p_rect, const Rect2 &p_src_rect, const Color &p_modulate = Color(1, 1, 1), bool p_transpose = false, const Ref<Texture> &p_normal_map = Ref<Texture>(), bool p_clip_uv = true) const;
 	virtual bool get_rect_region(const Rect2 &p_rect, const Rect2 &p_src_rect, Rect2 &r_rect, Rect2 &r_src_rect) const;
+	virtual RefineRectResult refine_rect_region(Rect2 &r_dst_rect, Rect2 &r_src_rect) const { return ((get_width() | get_height()) == 0) ? REFINE_RECT_RESULT_NO_DRAW : REFINE_RECT_RESULT_DRAW; }
```

Comment:
Since `r_dst_rect`, `r_src_rect` are as well inputs in here, it might make sense to return "no draw" also in case any of these rects has zero size.

Reply:
Although this seems to make sense for `r_dst_rect` (zero size should presumably not draw), maybe zero size `src_rect` is valid? The UVs would be the same but it can still draw something. This is similar to the question of whether to not draw in the case of `Texture` derived classes like `GradientTexture`. :thinking:

Reply:
> Although this seems to make sense for `r_dst_rect` (zero size should presumably not draw), maybe zero size `src_rect` is valid? The UVs would be the same but it can still draw something. This is similar to the question of whether to not draw in the case of `Texture` derived classes like `GradientTexture`. thinking

Hmm, in theory someone could want to transform vertices in the vertex shader so I guess even in case of a zero-sized `r_dst_rect` not drawing it could be argued against. :thinking: Not sure whether doing so makes sense / why someone would do it.

But I agree, zero-sized `r_src_rect` makes more sense. Like someone might want to draw a single color rect by sampling the texture at specific UV (again not sure if doing so makes sense).

So might be safest to not add rect related conditions here.
 
> Makes me again wonder if we should just mirror the old behaviour for safety.

Regarding width/height checks it might be fine to remove them. I'd assume trying to draw kinda invalid (zero-sized) textures is rather not a common case, so not preventing it right here shouldn't affect performance much anyway. And given it's against 3.x it might be better to prioritize avoiding potential regressions here indeed.